nsqd/protocol_v2.go

```diff
@@ -796,6 +797,9 @@ func (p *protocolV2) PUB(client *clientV2, params [][]byte) ([]byte, error) {
 		return nil, protocol.NewFatalClientErr(err, "E_PUB_FAILED", "PUB failed "+err.Error())
 	}
 
+	p.ctx.nsqd.AddClient(client.ID, client)
```

Comment:
why do we add the client every time here, can't we do this once when _any_ client connects? This relates to my comments above about variable naming.